diffkemp/semdiff/result.py

```diff
@@ -44,6 +45,94 @@ class Entity:
         or a parameter.
         If it is a function, it contains the file of the function.
         """
+        class Callstack:
+            """
+            Callstack of called functions.
+            Calls is a list of dictionaries with keys: name, file, line.
+            """
+
+            def __init__(self, calls):
+                self.calls = calls
+
+            def __add__(self, other):
+                return Result.Entity.Callstack(self.calls + other.calls)
+
+            @classmethod
+            def from_edge_objects(cls, callstack):
+                """Creates a Callstack from callstack consisting
+                of Edge objects."""
+                return cls([{"name": call.target_name, "file": call.filename,
+                             "line": call.line}
+                            for call in callstack])
+
+            @classmethod
+            def from_simpll_yaml(cls, callstack):
+                """Creates a Callstack from a YAML representation of
+                a callstack (used in non-fun) diffs."""
+                return cls([{"name": call["function"], "file": call["file"],
+                             "line": call["line"]}
+                           for call in callstack])
+
+            def as_str_with_rel_paths(self, startpath):
+                """Returns callstack as string with relative paths
+                to startpath."""
+                return str(self).replace(
+                    os.path.join(os.path.abspath(startpath), ""), "")
+
+            def to_output_yaml_with_rel_path(self, startpath):
+                """Returns callstack as output YAML representation with
+                relative file paths to startpath"""
+                if self.calls is None:
+                    return []
+                return [{"name": call["name"],
+                         "file": call["file"].replace(os.path.join(
+                             os.path.abspath(startpath), ""), ""),
+                         "line": call["line"]}
+                        for call in self.calls]
+
+            def get_symbol_names(self, fun_name):
+                """
+                Returns tree sets of symbol names which appears in callstacks,
+                names does not contain kind.
+                First set contains function names, second macro names
+                and third type names.
+                :param fun_name: Name of function where the callstack starts.
+
+                Note: type names contains tuples (type name, name of function
+                in which it is used)
```

Comment:
The comment is quite hard to understand. What about something like this:
```suggestion
                Returns a tuple containing three sets of symbol names which appear 
                in callstacks:
                - function names (set of strings)
                - macro names (set of strings)
                - type names (set of string pairs (type name, user function name))
                      
                :param fun_name: Name of the first callstack function.
```

Reply:
Thanks, I used most of your suggestion, but 
- for type names I edited the description of the second  of the pair to `name of function in which the type is used) `
- for param description I edited it to `Name of compared function`.
I have also renamed the param from `fun_name` to `compared_fun`, I think it better describe the param.